### PR TITLE
New Update

### PR DESCRIPTION
**Pull request by ekiuse**

### CODE REVIEW

File: javascript/src/controllers/phrese.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    DefaultResponseData,
+    ErrorResponseData,
+} from "../interfaces/utils.js";
+import {
+    PhraseListResponseData,
+    PhraseAddRequestData,
+    PhraseViewResponseData,
+} from "../interfaces/phrase.js";
+
+export class ControllerAyah {
```
Comment (Silence0ne) [inline javascript/src/controllers/phrese.ts:14]: Please rename it to **Phrase**

File: javascript/src/controllers/user.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    DefaultResponseData,
+    ErrorResponseData,
+} from "../interfaces/utils.js";
+import {
+    UserListItem,
+    UserAddRequestData,
+    UserListResponseData,
+} from "../interfaces/user.js";
+
+export class ControllerAyah {
```
Comment (Silence0ne) [inline javascript/src/controllers/user.ts:14]: Rename **ControllerAyah** to **ControllerUser**

**Review by al-abd — APPROVED** [review]

File: javascript/src/interfaces/profile.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+}
+export type ProfileListResponseData = ProfileListItem[];
+
+//profile Edit
+export interface ProfileEditRequestData {
+    username: string;
+    first_name: string | null;
+    last_name: string | null;
+    birthday: string | null;
+    profile_image: string | null;
```
Comment (Silence0ne) [inline javascript/src/interfaces/profile.ts:20]: Please add the **language** field to the `ProfileEditRequestData`

File: javascript/src/controllers/profile.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    RequestConfig,
+    DefaultResponseData,
+    ErrorResponseData,
+} from "../interfaces/utils.js";
+import {
+    ProfileListItem,
+    ProfileAddRequestData,
+} from "../interfaces/profile.js";
+
+export class ControllerAyah {
```
Comment (Silence0ne) [inline javascript/src/controllers/profile.ts:13]: Please rename it to `controllerProfile`

File: javascript/src/interfaces/profile.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+//profie List
+export interface ProfileListItem {
```
Comment (Silence0ne) [inline javascript/src/interfaces/profile.ts:2]: Please do not export **`ProfileListItem`**. Instead, use and import **`ProfileListResponseData`**, in the **ControllerProfile** and **index**.

 ```suggestion
interface ProfileListItem {
```

**Review by al-abd — APPROVED** [review]

**Review by Silence0ne — APPROVED** [review]

**Review by Silence0ne — APPROVED** [review]

**Pull request merged**